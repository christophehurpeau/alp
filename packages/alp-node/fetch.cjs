'use strict';

global.fetch = require('node-fetch');

module.exports = global.fetch;
