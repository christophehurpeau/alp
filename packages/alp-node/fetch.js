// eslint-disable-next-line no-restricted-exports
export { default } from './dist/fetch-node14.mjs';
