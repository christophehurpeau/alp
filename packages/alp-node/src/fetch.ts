import nodeFetch from 'node-fetch';

declare global {
  // eslint-disable-next-line vars-on-top, no-var
  var fetch: typeof nodeFetch;
}

global.fetch = nodeFetch;

// eslint-disable-next-line unicorn/prefer-export-from
export default nodeFetch;
