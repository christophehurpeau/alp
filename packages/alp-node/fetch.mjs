import fetch from 'node-fetch';

global.fetch = fetch;

// eslint-disable-next-line unicorn/prefer-export-from
export default fetch;
