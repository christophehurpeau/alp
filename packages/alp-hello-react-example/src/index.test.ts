// eslint-disable-next-line import/no-extraneous-dependencies
import fetch from 'node-fetch';
import type { Daemon } from 'springbokjs-daemon';
import { createDaemon } from 'springbokjs-daemon';

describe('test hello server', () => {
  let daemon: Daemon;

  beforeAll(async () => {
    daemon = createDaemon({
      command: 'node',
      cwd: new URL('../', import.meta.url).pathname,
      args: ['build/index.mjs', '--port', 5555, '--version', 'test'],
    });
    await daemon.start();
  });

  afterAll(async () => {
    await daemon.stop();
  });

  test('hello without name', async () => {
    const result = await fetch('http://localhost:5555/').then((res) =>
      res.text(),
    );
    expect(result).toMatchSnapshot();
  });

  test('hello with name', async () => {
    const result = await fetch('http://localhost:5555/?name=Chris').then(
      (res) => res.text(),
    );
    expect(result).toMatchSnapshot();
  });
});
