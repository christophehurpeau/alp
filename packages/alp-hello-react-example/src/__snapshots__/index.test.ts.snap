// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test hello server hello with name 1`] = `"<!DOCTYPE html><html><head><meta charSet=\\"utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width\\"/><title>Hello World!</title><meta name=\\"next-head-count\\" content=\\"3\\"/><noscript data-n-css=\\"\\"></noscript><script defer=\\"\\" nomodule=\\"\\" src=\\"/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js\\"></script><script src=\\"/_next/static/chunks/webpack-8b0bd4047513e2a7.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/framework-39ef0a24cced4e18.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/main-ad5db7eeb2a1f0a2.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/pages/_app-bd3252abcf2c6daa.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/2edb282b-b63b89b6504edc29.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/437-21e7f79178fb0e28.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/pages/index-026448f69e198dfd.js\\" defer=\\"\\"></script><script src=\\"/_next/static/test-build-id/_buildManifest.js\\" defer=\\"\\"></script><script src=\\"/_next/static/test-build-id/_ssgManifest.js\\" defer=\\"\\"></script></head><body><div id=\\"__next\\" data-reactroot=\\"\\"><div class=\\"css-1dbjc4n r-13awgt0\\"><div><div>Hello World !</div><div class=\\"css-1dbjc4n r-1awozwy r-6koalj r-1777fci r-1qd3jhq\\"><div class=\\"css-1dbjc4n r-hnxvie\\"><div dir=\\"auto\\" class=\\"css-901oao r-1njgeiw r-1i10wst r-13uqrnb r-16dba41 r-oxtfae r-dhbnww\\">Hello World!</div><div class=\\"css-1dbjc4n r-1awozwy r-g944k3 r-z2wwpe r-rs99b7 r-18u37iz r-1dqxon3\\"><input type=\\"text\\" autoComplete=\\"off\\" value=\\"\\" autoCapitalize=\\"sentences\\" autoCorrect=\\"on\\" dir=\\"auto\\" spellcheck=\\"true\\" class=\\"css-11aywtz r-1niwhzg r-1njgeiw r-113qch9 r-13awgt0 r-1enofrn r-13uqrnb r-16dba41 r-1pi2tsx r-dhbnww r-10paoce r-xd6kpl r-f727ji r-j2kj52 r-tskmnb r-uek9y r-13qz1uu\\" style=\\"outline:none\\" id=\\"undefined-input\\"/></div></div></div><div><p>You clicked <!-- -->0<!-- --> times</p><button type=\\"button\\">Click me</button></div><div><button type=\\"button\\">Click me to throw</button><button type=\\"button\\">Click me to log error</button></div></div></div></div><script id=\\"__NEXT_DATA__\\" type=\\"application/json\\">{\\"props\\":{\\"pageProps\\":{}},\\"page\\":\\"/\\",\\"query\\":{},\\"buildId\\":\\"test-build-id\\",\\"nextExport\\":true,\\"autoExport\\":true,\\"isFallback\\":false,\\"scriptLoader\\":[]}</script></body></html>"`;

exports[`test hello server hello without name 1`] = `"<!DOCTYPE html><html><head><meta charSet=\\"utf-8\\"/><meta name=\\"viewport\\" content=\\"width=device-width\\"/><title>Hello World!</title><meta name=\\"next-head-count\\" content=\\"3\\"/><noscript data-n-css=\\"\\"></noscript><script defer=\\"\\" nomodule=\\"\\" src=\\"/_next/static/chunks/polyfills-c67a75d1b6f99dc8.js\\"></script><script src=\\"/_next/static/chunks/webpack-8b0bd4047513e2a7.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/framework-39ef0a24cced4e18.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/main-ad5db7eeb2a1f0a2.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/pages/_app-bd3252abcf2c6daa.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/2edb282b-b63b89b6504edc29.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/437-21e7f79178fb0e28.js\\" defer=\\"\\"></script><script src=\\"/_next/static/chunks/pages/index-026448f69e198dfd.js\\" defer=\\"\\"></script><script src=\\"/_next/static/test-build-id/_buildManifest.js\\" defer=\\"\\"></script><script src=\\"/_next/static/test-build-id/_ssgManifest.js\\" defer=\\"\\"></script></head><body><div id=\\"__next\\" data-reactroot=\\"\\"><div class=\\"css-1dbjc4n r-13awgt0\\"><div><div>Hello World !</div><div class=\\"css-1dbjc4n r-1awozwy r-6koalj r-1777fci r-1qd3jhq\\"><div class=\\"css-1dbjc4n r-hnxvie\\"><div dir=\\"auto\\" class=\\"css-901oao r-1njgeiw r-1i10wst r-13uqrnb r-16dba41 r-oxtfae r-dhbnww\\">Hello World!</div><div class=\\"css-1dbjc4n r-1awozwy r-g944k3 r-z2wwpe r-rs99b7 r-18u37iz r-1dqxon3\\"><input type=\\"text\\" autoComplete=\\"off\\" value=\\"\\" autoCapitalize=\\"sentences\\" autoCorrect=\\"on\\" dir=\\"auto\\" spellcheck=\\"true\\" class=\\"css-11aywtz r-1niwhzg r-1njgeiw r-113qch9 r-13awgt0 r-1enofrn r-13uqrnb r-16dba41 r-1pi2tsx r-dhbnww r-10paoce r-xd6kpl r-f727ji r-j2kj52 r-tskmnb r-uek9y r-13qz1uu\\" style=\\"outline:none\\" id=\\"undefined-input\\"/></div></div></div><div><p>You clicked <!-- -->0<!-- --> times</p><button type=\\"button\\">Click me</button></div><div><button type=\\"button\\">Click me to throw</button><button type=\\"button\\">Click me to log error</button></div></div></div></div><script id=\\"__NEXT_DATA__\\" type=\\"application/json\\">{\\"props\\":{\\"pageProps\\":{}},\\"page\\":\\"/\\",\\"query\\":{},\\"buildId\\":\\"test-build-id\\",\\"nextExport\\":true,\\"autoExport\\":true,\\"isFallback\\":false,\\"scriptLoader\\":[]}</script></body></html>"`;
